io-engine/src/bdev/nexus/nexus_bdev.rs

```diff
@@ -602,6 +632,68 @@ impl<'n> Nexus<'n> {
         self.io_subsystem_mut().resume().await
     }
 
+    pub async fn shutdown(mut self: Pin<&mut Self>) -> Result<(), Error> {
+        // Check if nexus is not already shut down and mark nexus
+        // as being shutdown, preserving current nexus state.
+        let prev_state = {
+            let mut s = self.state.lock();
+
+            if *s == NexusState::Shutdown {
+                return Err(Error::OperationNotAllowed {
```

Comment:
Should this be a different error case than on `check_nexus_operation` ?

Reply:
Good point, we can put it here too.

Reply:
> what does this get mapped back up to via grpc so control plane can determine the call failed because we were already shutdown?

IIRC, as of now it will be mapped to 'Status::internal' error code.